pkg/run/apply_manifests_test.go

```diff
@@ -0,0 +1,92 @@
+package run_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fluxcd/flux2/pkg/manifestgen/install"
+	"github.com/fluxcd/pkg/ssa"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/pkg/logger/loggerfakes"
+	"github.com/weaveworks/weave-gitops/pkg/run"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// mock ssa.ResourceManager
+type mockResourceManagerForApply struct {
```

Comment:
Most of our mocks use counterfeiter - is there a reason it was easier here to write your own mock?

Reply:
Followed @chanwits's example with the mocking client methods, just reworked it a bit for the ResourceManager.

Thanks for your suggestion, will have a look at the counterfeiter.